pkg/apis/v1beta1/labels.go

```diff
@@ -128,7 +130,7 @@ func IsRestrictedNodeLabel(key string) bool {
 	return RestrictedLabels.Has(key)
 }
 
-func getLabelDomain(key string) string {
+func GetLabelDomain(key string) string {
```

Comment:
```suggestion
func getLabelDomain(key string) string {
```

Any reason that this needs to be a public method now?

Reply:
I have reused that function in karpenter itself https://github.com/aws/karpenter/pull/5122/files#diff-25302a63d425cf7ccce03fb744105007fed5dd3672e408875d41d950ed5e6970L159
Karpenter removes node-restriction.kubernetes.io labels from user-data in lanch template, but only as prefix.